### PR TITLE
Add AI Activity Tutorial

### PR DESCRIPTION
**Pull request by nicholasforeman**
# Description

Add new AI Tutorial based around Tic Tac Toe.

## Type of change

- [x] New content (non-breaking change which adds functionality)

### CODE REVIEW

**Review by StanzillaManticore — CHANGES_REQUESTED** [review]
One thing that is missing is adding the page to the nav menu. In both `mkdocs.yml` and `mkdocs-prod.yml`

**Comment by nicholasforeman** [comment]
> One thing that is missing is adding the page to the nav menu. In both `mkdocs.yml` and `mkdocs-prod.yml`

Added (AI Activities). Let me know if you want it under a different name.

File: src/tutorials/ai_tic_tac_toe.md [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+if isPlayerTurn then
+    activity.priority = 0
+    return
+end
+```
+
+Finally, select and verify there is a valid board position to capture.
+
+```lua
+-- If there is no valid capture position then we need to attempt a capture, otherwise, keep it above Idle priority
```
Comment (StanzillaManticore) [inline src/tutorials/ai_tic_tac_toe.md]: I would split to comment to two lines so it does not make the code box scroll on the max width version of the page.

File: src/tutorials/ai_tic_tac_toe.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+This concludes our `tick()` implementation: we set a "higher than idle" priority if the game is not over, if it is the AI's turn, and if a board position has been successfully determined; otherwise the priority is set to `0`, indicating that there is no actions to make.
+
+### TickHighestPriority Implementation
+
+This method is only executed when the **Capture Activity** has the highest priority, meaning its priority is higher than the **Idle Activity** priority (`100`). All there is to be done here is activate the trigger for the game board position. For aesthetics, it is nice to wait a second or two so it feels like the AI is responding at a comfortable pace.
+
+Add the following to the **Capture Activity** `tickHighestPriority()`:
+
+```lua
+-- serverUserData contains extra information about the activity that is user-defined and accessible to the server
```
Comment (StanzillaManticore) [inline src/tutorials/ai_tic_tac_toe.md]: same as above

File: src/tutorials/ai_tic_tac_toe.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+local activityData = activity.serverUserData
+
+-- Get the amount of time that has passed since the board position has been chosen
+local elapsedTime = activity.elapsedTime
+
+-- If the elapsed time is greater than 2 seconds, activate the trigger and set the capturePosition to be negative
```
Comment (StanzillaManticore) [inline src/tutorials/ai_tic_tac_toe.md]: same as above

**Review by StanzillaManticore — DISMISSED** [review]
small nits about comments, also apply to the `Final Code` section but I did not add markers there again.

**Review by StanzillaManticore — APPROVED** [review]
Content review I'll leave to @slinkous

**Comment by coreslinkous** [comment]
Can we rename to ai_activity_tutorial.md? Otherwise good to go.

**Pull request merged**